Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/ConnectionManagerBean.java

```diff
@@ -0,0 +1,273 @@
+ *       copyright notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the documentation
+ *       and/or other materials provided with the distribution.
+ *     * Neither the name of the United States Government nor the
+ *       names of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE UNITED STATES GOVERNMENT BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package gov.hhs.fha.nhinc.admingui.managed;
+
+import gov.hhs.fha.nhinc.admingui.application.EndpointManagerCache;
+import gov.hhs.fha.nhinc.admingui.model.ConnectionEndpoint;
+import gov.hhs.fha.nhinc.admingui.services.PingService;
+import gov.hhs.fha.nhinc.admingui.services.impl.PingServiceImpl;
+import gov.hhs.fha.nhinc.connectmgr.ConnectionManagerCache;
+import gov.hhs.fha.nhinc.connectmgr.ConnectionManagerException;
+import gov.hhs.fha.nhinc.nhinclib.NhincConstants;
+import gov.hhs.fha.nhinc.properties.PropertyAccessException;
+import gov.hhs.fha.nhinc.properties.PropertyAccessor;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import javax.annotation.PostConstruct;
+import javax.faces.bean.ManagedBean;
+import javax.faces.bean.ViewScoped;
+import org.apache.http.impl.cookie.DateUtils;
+import org.apache.log4j.Logger;
+import org.uddi.api_v3.BindingTemplate;
+import org.uddi.api_v3.BusinessEntity;
+import org.uddi.api_v3.BusinessService;
+import org.uddi.api_v3.CategoryBag;
+import org.uddi.api_v3.Contact;
+import org.uddi.api_v3.KeyedReference;
+
+/**
+ *
+ * @author jassmit
+ */
+@ManagedBean(name = "connectionManagerBean")
+@ViewScoped
+public class ConnectionManagerBean {
+
+    private static final Logger LOG = Logger.getLogger(ConnectionManagerBean.class);
+    
+    private final HashMap<String, BusinessEntity> externalEntities = new HashMap<String, BusinessEntity>();
+    private final List<String> entityNames = new ArrayList<String>();
+    
+    //TODO Display local entity endpoints (internal and external).
```

Comment:
It is out of scope as discussed tomorrow.  We’re only displaying the external entities endpoints for pinging.  Local endpoints (internal and external) will be covered in a future release, possibly 4.5.

From: Christopher May [mailto:notifications@github.com]
Sent: Tuesday, January 13, 2015 11:05 AM
To: CONNECT-Solution/CONNECT
Cc: Smith, Jason A (CGI Federal)
Subject: Re: [CONNECT] FHAC-94 Admin GUI Connection Manager (#994)


In Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/ConnectionManagerBean.java<https://github.com/CONNECT-Solution/CONNECT/pull/994#discussion-diff-22871700>:

> +import org.uddi.api_v3.KeyedReference;

> +

> +/**

> + *

> + * @author jassmit

> + */

> +@ManagedBean(name = "connectionManagerBean")

> +@ViewScoped

> +public class ConnectionManagerBean {

> +

> +    private static final Logger LOG = Logger.getLogger(ConnectionManagerBean.class);

> +

> +    private final HashMap<String, BusinessEntity> externalEntities = new HashMap<String, BusinessEntity>();

> +    private final List<String> entityNames = new ArrayList<String>();

> +

> +    //TODO Display local entity endpoints (internal and external).

Just double checking, this TODO is out of scope for 4.4.1?

—
Reply to this email directly or view it on GitHub<https://github.com/CONNECT-Solution/CONNECT/pull/994/files#r22871700>.